### PR TITLE
PUBDEV-7631: HGLM Python Example Should Reference 'random_columns' by Indices

### PR DESCRIPTION
**Pull request by hannah-tillman**

### CODE REVIEW

File: h2o-docs/src/product/data-science/algo-params/random_columns.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+none
+
+Example
+~~~~~~~
+
+.. tabs::
+	.. code-tab:: r R
+
+			library(h2o)
```
Comment (mn-mikke) [inline h2o-docs/src/product/data-science/algo-params/random_columns.rst:23]: nit: indentation

File: h2o-docs/src/product/data-science/algo-params/random_columns.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                          rand_link = c("identity"), 
+                          training_frame = h2odata, 
+                          HGLM = TRUE, 
+                          random_columns = z, 
+                          calc_like = TRUE)
+        print(m11H2O)
+
+	.. code-tab:: python
+
+			import h2o
```
Comment (mn-mikke) [inline h2o-docs/src/product/data-science/algo-params/random_columns.rst:51]: nit: indentation

File: h2o-docs/src/product/data-science/algo-params/random_columns.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        # Set the response, predictor, and random columns
+        yresp <- "y"
+        xlist <- c("x1", "x3", "x5", "x6")
+        z <- c(1)
+
+        # Convert the "Device" column to a factor
+        h2odata$Device <- h2o.asfactor(h2odata$Device)
+
+        # Train and view the model
+        m11H2O <- h2o.glm(x = xlist,
```
Comment (mn-mikke) [inline h2o-docs/src/product/data-science/algo-params/random_columns.rst:38]: nit: maybe a better name for the variable

File: h2o-docs/src/product/data-science/algo-params/hglm.rst [hunk 4]
```diff
@@ -64,7 +64,7 @@ Example
         # Set the response, predictor, and random columns
         y = "y"
         x = ["x1","x3","x5","x6"]
-        z = "Device"
+        z = 0
```
Comment (mn-mikke) [inline h2o-docs/src/product/data-science/algo-params/hglm.rst:67]: nit: Maybe `z` could contain a list and directly assigned : `random_columns=z` as it's in R example.

**Comment by angela0xdata** [comment]
@hannah-tillman I made a couple of edits. But can you also add a parameter appendix entry for `rand_family`? Once added:

- Add a link to that in the GLM chapter parameters list
- Add `rand_family` to related parameters for hglm and random_columns

File: h2o-docs/src/product/data-science/algo-params/rand_family.rst [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+~~~~~~~~~~~
+
+Hierarchical GLM (HGLM) fits generalized linear models with random effects, where the random effect can come from a conjugate exponential-family distribution (for example, Gaussian). The ``rand_family`` option specifies the Random Family Component as an array to be used in GLM when ``hglm=True``. 
+
+**Note:** You must include one family for each random component. Only ``rand_family={"[gaussian"})`` is currently supported.
+
+Related Parameters
+~~~~~~~~~~~~~~~~~~
+
+- `HGLM <hglm.html>`__
```
Comment (angela0xdata) [inline h2o-docs/src/product/data-science/algo-params/rand_family.rst:17]: UGH, you're correct. Only uppercase HGLM is allowed.

File: h2o-docs/src/product/data-science/algo-params/random_columns.rst [hunk 6]
```diff
@@ -13,6 +13,7 @@ Related Parameters
 ~~~~~~~~~~~~~~~~~~
 
 - `hglm <hglm.html>`__
```
Comment (angela0xdata) [inline h2o-docs/src/product/data-science/algo-params/random_columns.rst:15]: This should be uppercase (sorry)

File: h2o-docs/src/product/data-science/glm.rst [hunk 7]
```diff
@@ -87,7 +87,7 @@ Defining a GLM Model
    -  If the family is **gamma**, the response must be numeric and continuous and positive (**Real** or **Int**).
    -  If the family is **tweedie**, the response must be numeric and continuous (**Real**) and non-negative.
 
--  **rand_family**: The Random Component Family specified as an array. You must include one family for each random component. Currently only ``rand_family={"[gaussisan]"}`` is supported.
```
Comment (angela0xdata) [inline h2o-docs/src/product/data-science/glm.rst:90]: Can you make sure that "HGLM" is uppercase?

File: h2o-docs/src/product/data-science/algo-params/hglm.rst [hunk 8]
```diff
@@ -19,6 +19,7 @@ Related Parameters
 ~~~~~~~~~~~~~~~~~~
```
Comment (angela0xdata) [inline h2o-docs/src/product/data-science/algo-params/hglm.rst:20]: Can you make the title of this topic uppercase?

File: h2o-docs/src/product/data-science/algo-params/rand_family.rst [hunk 9]
```diff
@@ -0,0 +1,77 @@
+``rand_family``
+---------------
+
+- Available in: GLM
+- Hyperparameter: no
+
+Description
+~~~~~~~~~~~
+
+Hierarchical GLM (HGLM) fits generalized linear models with random effects, where the random effect can come from a conjugate exponential-family distribution (for example, Gaussian). The ``rand_family`` option specifies the Random Family Component as an array to be used in GLM when ``hglm=True``.
```
Comment (angela0xdata) [inline h2o-docs/src/product/data-science/algo-params/rand_family.rst:10]: need uppercase HGLM here and in random_columns.rst
> ... when `HGLM=True`

**Pull request merged**